adapters/jms-adapter-generator/src/main/java/uk/gov/justice/raml/jms/core/JmsEndpointGenerator.java

```diff
@@ -0,0 +1,163 @@
+
+        ramlResourceModels.stream()
+                .map(resource -> templateModelFrom(resource, configuration))
+                .forEach(model -> writeToTemplateFile(model, jmsTemplate, outputDirectory));
+
+    }
+
+    private void writeToTemplateFile(final Attributes attributes, final String jmsTemplate,
+            final File outputDirectory) {
+        final File file = new File(outputDirectory, createJmsFilenameFrom(attributes.uri));
+        try {
+            write(file, render(jmsTemplate, attributes.attributesMap));
+        } catch (IOException e) {
+            throw new JmsEndpointGeneratorException(format(OUTPUT_FILE_GENERATION_ERROR, attributes.uri),
+                    e);
+        }
+    }
+
+    private Attributes templateModelFrom(final Resource resource, final GeneratorConfig configuration) {
+        final String uri = resource.getUri();
+        final HashMap<String, String> data = new HashMap<>();
+
+        data.put("PACKAGE_NAME", configuration.getBasePackageName());
+        data.put("CLASS_NAME", classNameOf(uri));
+        data.put("ADAPTER_TYPE", componentOf(uri).name());
+        data.put("DESTINATION_LOOKUP", destinationNameOf(uri));
+        data.put("MESSAGE_SELECTOR", messageSelectorsFrom(resource.getActions()));
+
+        return new Attributes(data, uri);
+    }
+
+    private String messageSelectorsFrom(final Map<ActionType, Action> actions) {
+        if (actions.isEmpty()) {
+            throw new JmsEndpointGeneratorException(ACTIONS_EMPTY_ERROR);
+        }
+        return format("'%s'", parse(actions.get(ActionType.POST)));
+    }
+
+    private String parse(final Action action) {
+        return action.getBody().keySet().stream().map(this::commandNameOf).collect(Collectors.joining("','"));
+    }
+
+    private String commandNameOf(final String mediaType) {
+        final Matcher m = MEDIA_TYPE_PATTERN.matcher(mediaType);
+        m.find();
+        return m.group(2);
+    }
+
+    private String destinationNameOf(final String uri) {
+        return uri.replaceAll("/", "");
+    }
+
+    private Component componentOf(final String uri) {
+        final String[] uriParts = uri.split("\\.");
+        return Component.valueOf(uriParts[uriParts.length - 1], uriParts[uriParts.length - 2]);
+    }
+
+    @SuppressWarnings("resource")
+    private String jmsListenerTemplate() {
+        try (final InputStream stream = getClass().getResourceAsStream(JMS_TEMPLATE_RESOURCE)) {
+            return new Scanner(stream, UTF_8).useDelimiter("\\A").next();
+        } catch (IOException e) {
+            throw new JmsEndpointGeneratorException(TEMPLATE_LOADING_ERROR, e);
+        }
+    }
+
+    private String createJmsFilenameFrom(final String uri) {
+        return classNameOf(uri) + FILENAME_POSTFIX;
+    }
+
+    private String classNameOf(final String uri) {
+        return toCamelCase(uri);
+    }
+
+    private String toCamelCase(final String s) {
```

Comment:
yes we've considered it but it does not fit the the bill here (we need to camel case dot and/or slash delimited string)